### DISCUSSION TITLE
How do you list names of open PBIX files?

### DISCUSSION THREAD

**Discussion by SharmaAntriksh** [DaxStudio/DaxStudio#1059] [General]
Hi! Wondering what's the C# code for listing out all the open files, I can get the port number from the **_msmdsrv.port.txt_** but can't understand how to get the File name. I looked into the DAX Studio solution and there is lots of code :D still searching meanwhile.

**Comment by dgosbell** [comment]
Technically you can't get the file name, what I get is the window title of the Power BI Desktop process. In the DaxStudio.UI project there is a Utils folder with a PowerBIHelper.cs file. That file has the code which scans for running msmdsrv processes. I get the open port from that process and then I look for a parent pbidesktop.exe process. Then I read the window title from the pbidesktop process.
